src/tokenizer.rs

```diff
@@ -85,14 +80,19 @@ where
             ')' => Some(Token::ParenClose),
             ',' => Some(Token::Comma),
             c if is_numberlike(c) => {
-                let number = c.to_string() + &self.read_until_whitespace().unwrap_or_default();
-                match number.trim_start_matches('+').parse::<T>() {
+                let mut number = self.read_until_whitespace().unwrap_or_default();
+                if c != '+' {
+                    // Prepend the character because the string likely has capacity.
+                    number.insert(0, c);
```

Comment:
I'm not sure how significant the cost is, but inserting at the beginning is typically slower.

Do you think it would be worth changing:
```
fn read_until_whitespace(&mut self) -> Option<String>
```

To something like:
```
fn read_until_whitespace(&mut self, string: String) -> String
```

And then :
```
let number = self.read_until_whitespace(c.to_string());
```

It might not be worth it...

Reply:
Indeed, inserting at the beginning is typically slower, however allocation is generally much, much slower than moving some characters.   At this point we know the strings are short and likely have some spare capacity.  Previously it was doing char.to_string(), which creates a string with a capacity of 1 that will immediately be re-allocated when the string returned by read_until_whitespace is appended. 

Your suggestion is worth considering.  Even if it's not faster it would keep the string building in the same place.